### PR TITLE
Fixed Node's prune_dev_dependencies build property based scenarios and added verification tests

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -35,18 +35,19 @@ fi
 # rename the folders back appropriately for the current build
 if [ -d $allModulesDirName ]
 then
+	echo
```
Comment (kichalla) [inline src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl]: Thanks to @arroyc for finding this issue and bringing it to my attention.

File: src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl [hunk 2]
```diff
@@ -115,27 +116,28 @@ npm pack
 
 if [ "$copyOnlyProdModulesToOutput" == "true" ]
 then
-	# Rename folder having all node_modules as we want only prod dependencies
-	# to be synced with destination directory
-	mv node_modules $allModulesDirName
+	echo
+	echo "Copy '$SOURCE_DIR/node_modules' with all dependencies to '$SOURCE_DIR/$allModulesDirName'..."
+	rsync -rtE --links "node_modules/" "$allModulesDirName" --delete
```
Comment (arroyc) [inline src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl]: are we deleting links/files in destination?

File: tests/Oryx.Integration.Tests/Node/NodeWebFrontEndAppTest.cs [hunk 3]
```diff
@@ -62,5 +63,93 @@ public async Task CanBuildAndRun_NodeWebFrontEndApp(string nodeVersion)
...
+                },
+                async (hostPort) =>
+                {
+                    var data = await _httpClient.GetStringAsync($"http://localhost:{hostPort}/");
+                    Assert.Contains("Say It Again", data);
+                });
+        }
+
+        [Fact]
+        public async Task CanBuildAndRun_NodeWebFrontEndApp_AfterRebuild_WhenPruneDevDependenciesIsTrue()
```
Comment (arroyc) [inline tests/Oryx.Integration.Tests/Node/NodeWebFrontEndAppTest.cs]: I would also add a test where NodePlatform.PruneDevDependenciesPropertyKey=false/blank just to ensure we are not break anything currently working

**Comment by arroyc** [comment]
thanks for the nightly build … I was about to ask you that :)

File: tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs [hunk 4]
```diff
@@ -122,5 +123,46 @@ public async Task Node_CreateReactAppSample_zippedNodeModules(string nodeVersion
...
+            var runAppScript = new ShellScriptBuilder()
+                .AddCommand($"oryx -appPath {appOutputDir} -bindPort {ContainerPort}")
+                .AddCommand(DefaultStartupFilePath)
+                .ToString();
+            var buildScript = new ShellScriptBuilder()
+               .AddCommand(
+                $"oryx build {appDir} -i /tmp/int -o /tmp/out --platform nodejs " +
+                $"--platform-version {nodeVersion} -p {NodePlatform.CompressNodeModulesPropertyKey}=zip" +
+                $" -p {NodePlatform.PruneDevDependenciesPropertyKey}=true")
+               .AddCommand($"cp -rf /tmp/out/* {appOutputDir}")
```
Comment (arroyc) [inline tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs]: why not using -o {appOutputDir} directly

File: src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl (hunk 2 above)
Comment (kichalla) [reply src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl]: We are only deleting files/links in the target directory that are not present in source directory

File: tests/Oryx.Integration.Tests/Node/NodeWebFrontEndAppTest.cs (hunk 3 above)
Comment (kichalla) [reply tests/Oryx.Integration.Tests/Node/NodeWebFrontEndAppTest.cs]: Sure, sounds good. added one now.

File: tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs (hunk 4 above)
Comment (kichalla) [reply tests/Oryx.Integration.Tests/Node/NodeTestUsingZippedNodeModules.cs]: This is related to rsync errors that we see especially on windows machines. This is a workaround for it.

**Pull request merged**